lib/Reduce/Tester.cpp

```diff
@@ -131,12 +133,22 @@ void TestCase::ensureFileOnDisk() {
 
     // Write to the output.
     file = std::make_unique<llvm::ToolOutputFile>(filepath, fd);
-    module.print(file->os());
-    file->os().close();
-    if (file->os().has_error())
-      llvm::report_fatal_error(llvm::Twine("Error emitting the IR to file `") +
-                                   filepath + "`",
-                               false);
+    if (tester.emitBytecode) {
+      if (failed(writeBytecodeToFile(
+              module, file->os(),
+              mlir::BytecodeWriterConfig(getCirctVersion()))))
+        llvm::report_fatal_error(
+            llvm::Twine("Error emitting the IR to file `") + filepath + "`",
+            false);
+      file->os().close();
+    } else {
+      module.print(file->os());
+      file->os().close();
+      if (file->os().has_error())
+        llvm::report_fatal_error(
+            llvm::Twine("Error emitting the IR to file `") + filepath + "`",
+            false);
```

Comment:
Isn't that check relevant to both branches of the if-stmt?

Reply:
Good catch. This will be fixed in commit: d9d3b095d